tests/sentry/api/endpoints/test_event_owners.py

```diff
@@ -0,0 +1,196 @@
+        assert resp.data["owners"][2]["id"] == "%d" % self.user3.id
+
+    def test_multiple_owners_order_matters(self):
+        rules = [
+            Rule(Matcher("path", "*.py"), [Owner("user", self.user.email)]),
+            Rule(Matcher("path", "*foo*"), [Owner("user", self.user2.email)]),
+            Rule(Matcher("path", "*"), [Owner("user", self.user3.email)]),
+        ]
+        rules.reverse()
+
+        ProjectOwnership.objects.create(
+            project_id=self.project.id, schema=dump_schema(rules), fallthrough=True
+        )
+
+        event1 = self.store_event(
+            data={"stacktrace": {"frames": [{"filename": "foo.py"}]}}, project_id=self.project.id
+        )
+
+        self.path = reverse(
+            "sentry-api-0-event-owners",
+            kwargs={
+                "organization_slug": self.organization.slug,
+                "project_slug": self.project.slug,
+                "event_id": event1.event_id,
+            },
+        )
+
+        resp = self.client.get(self.path)
+        assert resp.status_code == 200
+        assert len(resp.data["owners"]) == 3
+        assert resp.data["owners"][0]["id"] == "%d" % self.user3.id
+        assert resp.data["owners"][1]["id"] == "%d" % self.user2.id
+        assert resp.data["owners"][2]["id"] == "%d" % self.user.id
+
+    def test_owners_of_different_types_ordered_correctly(self):
+        rules = [
+            Rule(Matcher("path", "*.py"), [Owner("user", self.user.email)]),
+            Rule(Matcher("path", "*.py"), [Owner("team", self.team3.slug)]),
+            Rule(Matcher("path", "*foo*"), [Owner("user", self.user2.email)]),
+            Rule(Matcher("path", "*.py"), [Owner("team", self.team2.slug)]),
+            Rule(Matcher("path", "*"), [Owner("user", self.user3.email)]),
+            Rule(Matcher("path", "*.py"), [Owner("team", self.team.slug)]),
+        ]
+
+        ProjectOwnership.objects.create(
+            project_id=self.project.id, schema=dump_schema(rules), fallthrough=True
+        )
+
+        event1 = self.store_event(
+            data={"stacktrace": {"frames": [{"filename": "foo.py"}]}}, project_id=self.project.id
+        )
+
+        self.path = reverse(
+            "sentry-api-0-event-owners",
+            kwargs={
+                "organization_slug": self.organization.slug,
+                "project_slug": self.project.slug,
+                "event_id": event1.event_id,
+            },
+        )
+
+        resp = self.client.get(self.path)
+        assert resp.status_code == 200
+        assert len(resp.data["owners"]) == 6
+        assert resp.data["owners"][0]["id"] == "%d" % self.user.id
+        assert resp.data["owners"][1]["id"] == "%d" % self.team3.id
+        assert resp.data["owners"][2]["id"] == "%d" % self.user2.id
+        assert resp.data["owners"][3]["id"] == "%d" % self.team2.id
+        assert resp.data["owners"][4]["id"] == "%d" % self.user3.id
+        assert resp.data["owners"][5]["id"] == "%d" % self.team.id
```

Comment:
Is it worth checking `type` here as well, just in case of id conflict?

Reply:
Yeah that could happen.

Reply:
This was a good point, it surfaced a bug.